### PR TITLE
Fixed bug when searching for joined entities of a doctrine-join-descriptor

### PR DESCRIPTION
**Pull request by turbo-ele**

### CODE REVIEW

File: src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php [hunk 1]
```diff
@@ -230,6 +230,28 @@ protected function getJoins()
...
+     */
+    protected function getAllFields($onlyReturnFilterFields = false)
+    {
+        $fields = array_merge(
+            $this->searchFields,
+            $this->sortFields,
+            $this->getUniqueExpressionFieldDescriptors($this->expressions)
+        );
+
+        if ($onlyReturnFilterFields !== true) {
```
Comment (wachterjohannes) [inline src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php]: move this up and return the array_merge result!

File: src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php [hunk 2]
```diff
@@ -230,6 +230,28 @@ protected function getJoins()
...
     /**
+     * Returns all FieldDescriptors that were passed to list builder.
+     *
+     * @param bool $onlyReturnFilterFields Define if only filtering FieldDescriptors should be returned
+     *
+     * @return AbstractDoctrineFieldDescriptor[]
+     */
+    protected function getAllFields($onlyReturnFilterFields = false)
+    {
+        $fields = array_merge(
```
Comment (wachterjohannes) [inline src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php]: like this:

```
if(...){return array_merge(...);}

return array_merge(...);
```

**Comment by wachterjohannes** [comment]
@turbo-ele ACK i will merge it when tests running (:

**Comment by turbo-ele** [comment]
@wachterjohannes nice ;)

**Pull request merged**